prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/migrationtoolkit/GetAnalysisTask.java

```diff
@@ -0,0 +1,120 @@
+				WorkParameter.builder().key("serverURL").type(WorkParameterType.TEXT).optional(true).description(
+						"Base URL of the MTA instance - e.g https://mta-openshift-mta.app.clustername.clusterdomain")
+						.build(),
+				WorkParameter.builder().key("bearerToken").type(WorkParameterType.TEXT).optional(true)
+						.description("Bearer token to authenticate server requests").build());
+	}
+
+	/**
+	 * @param workContext optional context values: serverURL, and bearerToken for the
+	 * mtaClient.
+	 */
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		if (mtaClient == null) {
+			serverUrl = URI.create(getOptionalParameterValue(workContext, "serverURL", null));
+			var bearerToken = getOptionalParameterValue(workContext, "bearerToken", null);
+			if (serverUrl == null) {
+				log.error(
+						"serverURL is empty. Either pass it while creating the instance of the task or in the context");
+				return new DefaultWorkReport(WorkStatus.FAILED, workContext);
+			}
+			mtaClient = new MTAClient(serverUrl, bearerToken);
+		}
+
+		int taskGroupID;
+		try {
+			taskGroupID = Integer.parseInt(getRequiredParameterValue(workContext, "taskGroupID"));
+		}
+		catch (MissingParameterException | NumberFormatException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		Result<TaskGroup> result = mtaClient.get(taskGroupID);
+
+		if (result == null) {
+			// unexpected
+			return new DefaultWorkReport(WorkStatus.FAILED, new WorkContext(),
+					new IllegalStateException("MTA client returned empty result with no error."));
+		}
+		else if (result instanceof Result.Failure<TaskGroup> failure) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, failure.t());
+		}
+		else if (result instanceof Result.Success<TaskGroup> success) {
+			if ("Ready".equals(success.value().state()) && success.value().tasks() != null
+					&& success.value().tasks()[0].state().equals("Succeeded")) {
+				String reportURL = String.format("%s/hub/applications/%d/bucket/%s", serverUrl,
+						success.value().tasks()[0].application().id(), success.value().data().output());
+				sendEmail(reportURL, getOptionalParameterValue(workContext, "email", null));
+				addParameter(workContext, "reportURL", reportURL);
+				return new DefaultWorkReport(WorkStatus.COMPLETED, workContext);
+			}
+			else if ("Failed".equals(success.value().state())) {
+				return new DefaultWorkReport(WorkStatus.FAILED, workContext,
+						new Throwable("The underlying task failed, the report will not be ready"));
+			}
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Throwable("The report is not ready yet."));
+		}
+		throw new IllegalArgumentException();
+	}
+
+	private void sendEmail(String reportURL, String recipient) {
+		if (recipient == null) {
+			return;
+		}
+		messageConsumer.accept(new Message(recipient, "parodos-task-notificaion+mailtrap@redhat.com",
```

Comment:
should the sender be specified here? not from the context / task params? or should it be replaced by a value from the config-server in the future?